boofuzz/socket_connection.py

```diff
@@ -159,12 +159,28 @@ def open(self):
 
         if self.server:
             self._sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            self._sock.bind((self.host, self.port))
+            try:
+                self._sock.bind((self.host, self.port))
+            except socket.error as e:
+                # Handle when socket is in TIME_WAIT state.
+                if e.errno == errno.EADDRINUSE:
+                    raise exception.BoofuzzOutOfAvailableSockets()
+                else:
+                    raise
 
             if self.proto == "tcp" or self.proto == "ssl":
                 self._serverSock = self._sock
-                self._serverSock.listen(1)
-                self._sock, addr = self._serverSock.accept()
+                try:
+                    self._serverSock.listen(1)
+                    self._sock, addr = self._serverSock.accept()
+                except socket.error as e:
+                    # When connection timeout expires, tear down the server socket so we can re-open it again after
+                    # restarting the target. Can't just call self.close() because self._sock isn't as expected for it.
```

Comment:
Without having tested it, I currently don't see why we can't use `self.close()`. Does `self._sock.close()` cause an exception?

Reply:
So I tested and actually self.close() works fine, so I changed it to that. I had assumed that since we are handling an error in listen() or accept() that self._sock would still be the same object as self._serversock, and further assumed the double close on that object would cause issues.

I guess it works fine so to help keep things consistent in the future using self.close() is a better idea, to avoid having close code in multiple places.